torch/csrc/distributed/rpc/future_message.cpp

```diff
@@ -56,6 +60,24 @@ void FutureMessage::fireCallbacks() {
   callbacks_.clear();
 }
 
+void FutureMessage::startTimer() {
+  futureStartTime_ = std::chrono::high_resolution_clock::now();
+}
+
+bool FutureMessage::checkTimeElapsed(
+    const std::chrono::seconds& timeoutSeconds) {
+  std::unique_lock<std::mutex> lock(mutex_);
+
+  if (completed_) {
+    return false;
+  }
+  const auto now = std::chrono::high_resolution_clock::now();
```

Comment:
Please use std::chrono::steady_clock from what I recall high_resolution_clock isn't monotonic and may be more expensive than steady_clock.